provisioning/roles/mangaki_source/tasks/main.yml

```diff
@@ -0,0 +1,101 @@
+---
+- name: Ensure project group exists.
+  group:
+    name: '{{ mangaki_source_group }}'
+    state: 'present'
+
+- name: Ensure project user exists.
+  user:
+    name: '{{ mangaki_source_user }}'
+    group: '{{ mangaki_source_group }}'
+    state: 'present'
+
+- name: Ensure python and virtualenv are installed and up-to-date.
+  apt:
+    name: '{{ item }}'
+    state: 'latest'
+  with_items:
+    - 'python3'
+    - 'python3-virtualenv'
+    - 'virtualenv'
+  become: true
+
+- name: Ensure virtualenv packages are up-to-date.
+  pip:
+    name: '{{ item }}'
+    state: 'latest'
+    virtualenv: '{{ mangaki_source_venv_path }}'
+    virtualenv_python: 'python3'
+  become: true
+  become_user: '{{ mangaki_source_user }}'
+  with_items:
+    - 'pip'
+    - 'setuptools'
+
+- name: Ensure packaged project is present.
+  copy:
+    src: '{{ mangaki_source_package_src }}'
+    dest: '{{ mangaki_source_package }}'
+    owner: '{{ mangaki_source_user }}'
+    group: '{{ mangaki_source_group }}'
+    mode: '0644'
+  when: 'mangaki_source_install_type == "copy"'
+
+- name: Ensure system-wide development dependencies are installed.
+  apt:
+    name: '{{ item }}'
+    state: 'latest'
+  become: true
+  when: 'mangaki_source_install_type == "develop"'
+  with_items: ['git', 'tmux']
+
+- name: Ensure debug-mode dependencies are installed.
+  pip:
+    requirements: '{{ mangaki_source_package }}/../requirements/dev.txt'
+    virtualenv: '{{ mangaki_source_venv_path }}'
+  when: 'mangaki_source_debug|bool'
+  become: true
+  become_user: '{{ mangaki_source_user }}'
+
+- name: Ensure project is installed.
+  pip:
+    name: '{{ mangaki_source_package }}'
+    state: 'latest'
+    extra_args: '{{ (mangaki_source_install_type == "develop")|ternary("-e", omit) }}'
+    virtualenv: '{{ mangaki_source_venv_path }}'
+  become: true
+  become_user: '{{ mangaki_source_user }}'
+  register: project_install
+
+- name: Request service reloading.
+  set_fact:
+    mangaki_needs_restart: true
+  when: project_install.changed
+
+- name: Ensure settings.ini is present.
+  template:
+    src: 'settings.ini.j2'
+    dest: '{{ mangaki_source_settings_path }}'
+    owner: '{{ mangaki_source_user }}'
+    group: '{{ mangaki_source_group }}'
+    mode: '0600'
+  register: settings_ini
+
+- name: Request service reloading.
+  set_fact:
+    mangaki_needs_restart: true
+  when: settings_ini.changed
+
+- name: Insert/update virtualenv configuration to bashrc.
+  blockinfile:
+    path: '{{ mangaki_source_home }}/.bashrc'
+    block: |
+      source {{ mangaki_source_venv_path }}/bin/activate
```

Comment:
I'm using `blockinfile` because I want to append to the .bashrc file, not modify it -- I don't know of a good way of doing that with a separate file (except for shipping that file over and sourcing it using `blockinfile`; but since those are only a few lines meant for development machines I felt it was not necessary).